### PR TITLE
[release/8.0] Numerics and Tensors backport

### PR DESCRIPTION
**Pull request by michaelgsharp**
Backporting the new BCL.Numerics and Tensors packages.

## Customer Impact
Adds in MathF support for older frameworks so people don't have to roll their own. This allows for us and for 3rd parties to remove duplicated code in places where we are doing this ourselves.

The Tensors code adds in some missing math features that will be able to be hardware accelerated increasing the performance.

## Testing
All new API's have automated testing.

## Risk
Risk should be minimal to none. the BCL package is a new package adding in MathF support to older frameworks. The Tensors package hasn't been shipped for many releases, so these changes shouldn't affect anyone. And since we aren't changing existing API's, new automated testing should be enough to mitigated any risks.

### CODE REVIEW

File: src/libraries/Microsoft.Bcl.Numerics/src/PACKAGE.md [hunk 1]
```diff
@@ -10,7 +10,22 @@ As of .NET Core 2.0 and .NET Standard 2.1, the C# language has support for math
 
 ## How to Use
 
-<!-- A compelling example on how to use this package with code, as well as any specific guidelines for when to use the package -->
```
Comment (ericstj) [inline src/libraries/Microsoft.Bcl.Numerics/src/PACKAGE.md:13]: Please remove the duplicate sample.

**Comment by ericstj** [comment]
I think you are missing the following commits:
https://github.com/dotnet/runtime/commit/2dfee0f3520520a331ff9ba1c63b0758150e8cf1
https://github.com/dotnet/runtime/commit/91ac6b3544d53fcf9e6962575813c1cd54aa27bd

**Comment by michaelgsharp** [comment]
Those commits have been added.

**Review by ericstj — APPROVED** [review]
Approve of this feature for RC2. @artl93  please give it your ack.

**Review by jeffhandley — APPROVED** [review]

**Review by artl93 — APPROVED** [review]
M2 approved.

**Review by artl93 — APPROVED** [review]

**Comment by carlossanlop** [comment]
@michaelgsharp the CI failed in this PR: some Tensors types are not found. Can you please take a look tomorrow, since it's affecting the release/8.0 branch?

**Comment by stephentoub** [comment]
> @michaelgsharp the CI failed in this PR: some Tensors types are not found. Can you please take a look tomorrow, since it's affecting the release/8.0 branch?

https://github.com/dotnet/runtime/pull/92269
https://github.com/dotnet/runtime/pull/92270

**Comment by ericstj** [comment]
I'll get this fixed ASAP.  Looks like a couple ifdefs are missing.

**Comment by stephentoub** [comment]
> I'll get this fixed ASAP. Looks like a couple ifdefs are missing.

I already put up the prs for it.

**Pull request merged**